### PR TITLE
Quartz beta dev layer status

### PR DESCRIPTION
**Pull request by manishk3189**
@doneill I have created a sample to demonstrate how to view the Layer Load Status. 
Can you please review the PR in reference to [#986](https://devtopia.esri.com/runtime/java-android-sdk/issues/986) and let me know of changes if required?

### CODE REVIEW

**Comment by doneill** [comment]
@manishk3189 This sample does not have a README. Please add this before I start reviewing.

**Comment by manishk3189** [comment]
Hi @doneill I have created a README.md for the sample. Can you please review the PR and let me know of changes if required?

File: display-layer-view-state/src/main/java/com/esri/arcgisruntime/sample/displaylayerviewstate/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        });
+    }
+
+    /**
+     * The method looks up the view status of the layer and returns a string which is displayed
+     *
+     * @param status View Status of the layer
+     * @return String equivalent of the status
+     */
+    protected String viewStatusString (String status) {
```
Comment (doneill) [inline display-layer-view-state/src/main/java/com/esri/arcgisruntime/sample/displaylayerviewstate/MainActivity.java]: Make declaration private

File: display-layer-view-state/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            android:textColor="@color/colorPrimaryDark"
+            android:textStyle="bold" />
+        <TextView
+            android:id="@+id/worldTimeZoneStatusView"
+            android:layout_width="125dp"
+            android:layout_height="wrap_content"
+            android:layout_alignParentEnd="true"
+            android:layout_alignParentRight="true"
+            android:layout_alignTop="@+id/worldTimeZoneView"
+            android:singleLine="true"
```
Comment (doneill) [inline display-layer-view-state/src/main/res/layout/activity_main.xml]: `android:singleLine` is deprecated, use `maxLines=1` here and on line 35 above.

File: display-layer-view-state/build.gradle [hunk 3]
```diff
@@ -0,0 +1,24 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion
+
+    defaultConfig {
+        applicationId "com.esri.arcgisruntime.sample.displaylayerviewstate"
+        minSdkVersion 16
+        targetSdkVersion 23
```
Comment (doneill) [inline display-layer-view-state/build.gradle]: Not targeting the latest version of Android 24.

File: display-layer-view-state/build.gradle [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    compile 'com.android.support:appcompat-v7:23.4.0'
```
Comment (doneill) [inline display-layer-view-state/build.gradle]: Test the newer version of the appcompat-v7 lib, 24.0.0

File: display-layer-view-state/src/main/java/com/esri/arcgisruntime/sample/displaylayerviewstate/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        mMapView.addLayerViewStateChangedListener(new LayerViewStateChangedListener() {
+            @Override
+            public void layerViewStateChanged(LayerViewStateChangedEvent layerViewStateChangedEvent) {
+
+                // get the layer which changed it's state
+                Layer layer = layerViewStateChangedEvent.getLayer();
+
+                // get the View Status of the layer
+                // View status will be either of ACTIVE, ERROR, LOADING, NOT_VISIBLE, OUT_OF_SCALE, UNKNOWN
+                String viewStatus = layerViewStateChangedEvent.getLayerViewStatus().iterator().next().toString();
```
Comment (doneill) [inline display-layer-view-state/src/main/java/com/esri/arcgisruntime/sample/displaylayerviewstate/MainActivity.java]: This line is not getting invoked when the World Time Zones layer changes resulting in a `viewStatus=LOADED` on line 103 even when it is not loaded.  Reference the screenshot below to show the issue in the app:

time-zone-active

**Comment by doneill** [comment]
@manishk3189 Initial review comments for you to address, biggest issue is with the **WorldTimeZones** layer not updating it's state.

**Comment by manishk3189** [comment]
@doneill After initial investigation, `layerViewStateChangedEvent.getLayer()` gets the name of the layer which just changed it view state. When the map is panned out to extent such that **WorldTimeZone** is not visible the status of the layer is not changed thus it is not reflected. However, if a minimum scale is set to the **WorldTimeZone** tiled layer, the status is reflected correctly in the TextView. I have made the said changes, do let me know if you still want create an issue and track the progress there.

**Comment by doneill** [comment]
@mani8177 I will accept your workaround and investigate the issue you raise.  Good catch!

**Pull request merged**